tree/ntuple/v7/src/RPageStorage.cxx

```diff
@@ -309,6 +309,25 @@ ROOT::Experimental::Detail::RPageSink::AddColumn(DescriptorId_t fieldId, const R
    return ColumnHandle_t{columnId, &column};
 }
 
+void ROOT::Experimental::Detail::RPageSink::FixUpColumnRepresentative(RFieldBase &field)
+{
+   if (GetWriteOptions().GetCompression() != 0)
+      return;
+   if (!field.HasDefaultColumnRepresentative())
+      return;
+   const auto &rep = field.GetColumnRepresentative();
+   if (rep == RFieldBase::ColumnRepresentation_t({EColumnType::kSplitReal64})) {
+      field.SetColumnRepresentative({EColumnType::kReal64});
+   } else if (rep == RFieldBase::ColumnRepresentation_t({EColumnType::kSplitReal32})) {
+      field.SetColumnRepresentative({EColumnType::kReal32});
+   } else if (rep == RFieldBase::ColumnRepresentation_t({EColumnType::kSplitInt64})) {
+      field.SetColumnRepresentative({EColumnType::kInt64});
+   } else if (rep == RFieldBase::ColumnRepresentation_t({EColumnType::kSplitInt32})) {
+      field.SetColumnRepresentative({EColumnType::kInt32});
+   } else if (rep == RFieldBase::ColumnRepresentation_t({EColumnType::kSplitInt16})) {
+      field.SetColumnRepresentative({EColumnType::kInt16});
+   }
+}
```

Comment:
I'd vote against the storage layer doing this kind of fixups.  Perhaps we can have `RFieldBase` fix its own encoding via a `OnConnectPageSink()` virtual call -- we already have `OnConnectPageSource()` (see [here](https://github.com/root-project/root/blob/658f6ae18d6fe93f686e315aa75739d549fa88bb/tree/ntuple/v7/inc/ROOT/RField.hxx#L193-L195), which we needed for registering I/O rules).

Reply:
Yep, I moved the fixup to the `RFieldBase` class although not with an `OnConnectPageSink()` callback.  I'm not sure if the current state is the best we can do.  But I also wasn't fully happy with the callback, which I found made this block of "uncompressed files used unsplit encoding by default" quite opaque behind many methods.